### DISCUSSION TITLE
Getting URL to download raw file from private repository

### DISCUSSION THREAD

**Discussion by SaintQd** [community/community#84909] [New to GitHub] [answered]
Is there a way to allow someone else to download specific file from private repository using only browser URL? Already generated personal access token but don't know where to use it.

**Comment by mzums** [comment]
Yes, there is. This question was already answered here:
https://stackoverflow.com/questions/18126559/how-can-i-download-a-single-raw-file-from-a-private-github-repo-using-the-comman
and shows both how to download it with tokens and only with the URL.

If this reply was helpful please mark it as the answer :)

**Comment by SaintQd** [comment]
> Yes, there is. This question was already answered here: https://stackoverflow.com/questions/18126559/how-can-i-download-a-single-raw-file-from-a-private-github-repo-using-the-comman and shows both how to download it with tokens and only with the URL.
> 
> If this reply was helpful please mark it as the answer :)

Is there a way to do this without "curl" command? I need to download a file using only browser URL, can't use any commands.

**Comment by akashdip2001** [comment]
1. **Create a Personal Access Token (PAT):**
   - Go to your GitHub account Settings.
   - Click on "Developer settings," then "Personal access tokens."
   - Generate a new token with the necessary permissions (e.g., repo).
   - Copy the token as you won't be able to see it again.

2. **Construct the Download URL:**
   - Use the following URL format:
     ```
     https://<username>:<token>@github.com/<owner>/<repo>/releases/download/<tag>/<filename>
     ```
     Replace `<username>`, `<token>`, `<owner>`, `<repo>`, `<tag>`, and `<filename>` with your GitHub username, PAT, repository owner, repository name, release tag, and the specific file you want to share.

3. **Share the URL:**
   - Provide the generated URL to the person who needs to download the file.

Remember to keep your PAT secure and only share the URL with trusted individuals, as it grants access to your repository.

**Reply by jsoref** [reply]
Doing this would allow anyone with the url to clone the repository which can't possibly be desirable.

**Comment by jsoref** [comment, marked as answer]
Personally, I'd just create a public repository with no content (possibly a readme) and then download the file from the private repository and upload it a an artifact/blob/something in the public repository.

**Reply by SaintQd** [reply]
That's a better idea, thanks.
